wiring/inc/spark_wiring_logging.h

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2016 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef SPARK_WIRING_LOGGING_H
+#define SPARK_WIRING_LOGGING_H
+
+#include <initializer_list>
+#include <vector>
```

Comment:
Definitely it's better to refactor this code, i.e. for cloud-enabled logging, which can be enabled at any time. Direct alternative is to use some simple wrapper over `realloc` instead of `std::vector`.